hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/replication/ReplicationManager.java

```diff
@@ -448,6 +436,153 @@ protected ContainerHealthResult processContainer(ContainerInfo containerInfo,
     return health;
   }
 
+  private ContainerHealthResult checkContainerState(
+      final ContainerInfo containerInfo, final ReplicationManagerReport report)
+      throws ContainerNotFoundException {
+    ContainerID containerID = containerInfo.containerID();
+    Set<ContainerReplica> replicas = containerManager
+        .getContainerReplicas(containerID);
+    HddsProtos.LifeCycleState state = containerInfo.getState();
+    ContainerHealthResult health =
+        new ContainerHealthResult.HealthyResult(containerInfo);
+
+    if (state == HddsProtos.LifeCycleState.OPEN) {
+      if (!isOpenContainerHealthy(containerInfo, replicas)) {
+        report.incrementAndSample(
+            HealthState.OPEN_UNHEALTHY, containerID);
+        eventPublisher.fireEvent(SCMEvents.CLOSE_CONTAINER, containerID);
+        return new ContainerHealthResult.UnHealthyResult(containerInfo);
+      }
+      return health;
+    }
+
+    /*
+     * If the container is in CLOSING state, the replicas can either
+     * be in OPEN or in CLOSING state. In both of this cases
+     * we have to resend close container command to the datanodes.
+     */
```

Comment:
Nit: I think these comments (this one and the one for "DELETING" state) would be better placed at the corresponding `handle...` methods.